lib/modules/platform/gitea/index.ts

```diff
@@ -295,20 +315,19 @@ const platform: Platform = {
     };
   },
 
-  async getRepos(): Promise<string[]> {
+  async getRepos(config?: AutodiscoverConfig): Promise<string[]> {
     logger.debug('Auto-discovering Gitea repositories');
     try {
-      const repos = await helper.searchRepos({
-        uid: botUserID,
-        archived: false,
-        ...(process.env.RENOVATE_X_AUTODISCOVER_REPO_SORT && {
-          sort: process.env.RENOVATE_X_AUTODISCOVER_REPO_SORT as RepoSortMethod,
-        }),
-        ...(process.env.RENOVATE_X_AUTODISCOVER_REPO_ORDER && {
-          order: process.env.RENOVATE_X_AUTODISCOVER_REPO_ORDER as SortMethod,
-        }),
+      if (!config?.topics) {
+        return await fetchRepositories();
+      }
+
+      let repos: string[] = [];
+      await p.map(config.topics, async (topic: string) => {
+        const r = await fetchRepositories(topic);
+        repos = [...new Set([...repos, ...r])];
```

Comment:
return repos from callback, do a flatMap ant had unique filtering after that.
will perform better, because we've less set and array creations.

use this for dedupe
https://github.com/renovatebot/renovate/blob/f47704f7d6560c180f6885f4934691d46f34fa1a/lib/util/array.ts#L32

Reply:
@viceice done